src/stores/my-account.ts

```diff
@@ -27,23 +49,35 @@ type Session = {
   isNewSecretKey?: boolean
 }
 type Actions = {
-  login: (
+  isAuthenticated?: boolean
+  loginAnonymously: (
     secretKey?: string,
     isInitialization?: boolean
   ) => Promise<string | false>
-  logout: () => void
+  initializeWeb3Auth: () => Promise<Web3Auth>
```

Comment:
this comment is related to the comment above regarding the web3auth and provider state, you should change this function to just like usual function outside of zustand

Reply:
Agreed, this is only called once when the user logs in and out. But the web3auth returned needs to be saved to the zustand store.

Or we'll have to create new instance when the actions are called.